cluster/src/main/java/com/hartwig/pipeline/resource/ResourceFilesFactory.java

```diff
@@ -9,13 +9,15 @@ public static ResourceFiles buildResourceFiles(final RefGenomeVersion version) {
     }
 
     public static ResourceFiles buildResourceFiles(final CommonArguments arguments) {
+
         ResourceFiles resourceFiles = buildResourceFiles(arguments.refGenomeVersion());
+
         if (arguments.refGenomeUrl().isPresent()) {
             resourceFiles = new OverriddenReferenceGenome(resourceFiles, arguments.refGenomeUrl().get());
         }
-        if (arguments.targetRegionsBedLocation().isPresent()) {
-            resourceFiles = new TargetRegionsEnabled(resourceFiles, arguments.targetRegionsBedLocation().get());
-        }
+
+        TargetRegionsFiles.parseArguments(resourceFiles, arguments);
```

Comment:
From a quick scan it's difficult to see what this is doing. The method is called parseArguments but doesn't return anything, so it must actually be mutating the arguments. Mutating arguments to a function is something we avoid on principle.